### ISSUE TITLE
"uploadFile" method doesn't return data object on iOS

### ISSUE THREAD

**Issue by mifkys** [silkimen/cordova-plugin-advanced-http#27]
After execution of UploadFile method it returns only 'headers' and 'status' keys. File has been uploaded but app don't receive 'data' key with data from server.

code example:
```
let seq = this.http.uploadFile(this.url + '/' + endpoint, {}, {}, file_uri, "photo");

    seq
      .then(response => {
        console.log("Response:", response);
      )
      .catch(err => {});
```

response log:
```
Object
headers: Object
  connection: "keep-alive"
  content-encoding: "gzip"
  content-type: "application/json; charset=utf-8"
  date: "Fri, 06 Oct 2017 21:23:37 GMT"
  server: "nginx/1.2.7"
  strict-transport-security: "max-age=15552000"
  transfer-encoding: "Identity"
  vary: "Authorization, Cookie"
status: 200
```

Same server but uploading with cordova-plugin-file-transfer. Response data exists in response object:
```
const fileTransfer: TransferObject = this.transfer.create();
    let transferOptions: FileUploadOptions = {
       fileKey: 'photo',
       fileName: file_uri.substr(file_uri.lastIndexOf('/') + 1)
    }
    fileTransfer.upload(file_uri, this.url + '/photo/', transferOptions).then(data => {
      let res = JSON.parse(data.response);
      if (res.success) {
        this.setProfilePhoto(res.photo);
      }
    }, err => {});
```

**Comment by Faisalali23** [comment]
On success completion response object is not passed that`s why it only returns status and header 
- find the file "CordovaHttpPlugin.m" in plugins folder
- find the method "(void)uploadFile:(CDVInvokedUrlCommand*)command" in above mentioned file
- Change the line in above mentioned method
[self handleSuccess:dictionary withResponse:(NSHTTPURLResponse*)task.response andData:nil];
to this:
[self handleSuccess:dictionary withResponse:(NSHTTPURLResponse*)task.response andData:responseObject];

![plugin-update](https://user-images.githubusercontent.com/14192872/32210580-47824faa-be30-11e7-93af-c0dde4d98947.png)

**Comment by silkimen** [comment]
I can confirm that this problem is a bug in iOS implementation. It will be fixed in v1.8.1.

**Comment by Faisalali23** [comment]
Yes it is bug while implementing for ios
